### ISSUE TITLE
The loss values from different GPUs are the same

### ISSUE THREAD

**Issue by yianzhongguo** [horovod/horovod#2047]
**Environment:**
1. Framework: (TensorFlow, Keras, PyTorch, MXNet)
2. Framework version:
3. Horovod version:
4. MPI version:
5. CUDA version:
6. NCCL version:
7. Python version:
8. OS and version:
9. GCC version:

**Checklist:**
1. Did you search issues to find if somebody asked this question before?
2. If your question is about hang, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/running.rst)?
3. If your question is about docker, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/docker.rst)?
4. Did you check if you question is answered in the [troubleshooting guide](https://github.com/horovod/horovod/blob/master/docs/troubleshooting.rst)?

**Your question:**
Please ask your question here.
I can use Horovod normally. However, today I find the outputs from all the GPUs are all the same. Since the inputs are different and the initializaions of the model are random, I should get different outputs for different GPUs. I don't know the reason and do you meet with it and is it normal?
![loss_output](https://user-images.githubusercontent.com/30361876/85270348-bbb79780-b4ab-11ea-9010-11dc0b94b581.png)

**Comment by yianzhongguo** [comment]
Sorry, I set a seed when processing data. When I delete the seed all work as expected.
